.gitignore

```diff
@@ -49,3 +49,6 @@ result
 /test/
 /iroha-java/
 /lcov.info
+
+/rust-toolchain.toml
+.cargo/config.toml
```

Comment:
Do we really want to ignore this files?

Reply:
I don't know _why_ we have been ignoring those files, but I've accomplished this by listing them `.git/info/exclude`, which works the same way as gitignore, but isn't committed.

I think it does make sense to commit them to ensure everybody uses the same toolchain we use in CI and to not make people enable lints feature by hand.

I also have a snippet that  enables the usage of `mold` there:

```toml
[target.x86_64-unknown-linux-gnu]
linker = "clang"
rustflags = ["-C", "link-arg=-fuse-ld=/usr/bin/mold"]
```

Reply:
> but I've accomplished this by listing them .git/info/exclude

Oh, thanks for the tip

Reply:
Ah, just forgot to remove them from `staged`
Will move to `.git/info/exclude`, although I would like to see them in the repo permanently anyway.

Reply:
Restored `.gitignore`.